### PR TITLE
 boards: arm: Add basic support for Silabs EFR32 SLWSTK6061A board 

### PR DESCRIPTION
**Pull request by chrta**
This pull request contains the following:

- SoC files for the Silabs EFR32FG1P
- UART and GPIO driver adapted for this device
- Simple board implementation for the EFR32-SLWSTK6061A board 
  - only GPIO and UART are implemented

### CODE REVIEW

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst [hunk 1]
```diff
@@ -0,0 +1,185 @@
+.. _efr32_slwstk6061a:
+
+EFR32-SLWSTK6061A
+#################
+
+Overview
+********
+
+The EFR32 Flex Gecko Wireless Starter Kit SLWSTK6061A contains a Wireless
+System-On-Chip from the EFR32FG family built on ARM® Cortex®-M4F processor
```
Comment (dbkinder) [inline boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: ``on an ARM``

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+- 2 user buttons and 2 LEDs
+- Si7021 Humidity and Temperature Sensor
+- On-board Segger J-Link USB and Ethernet debugger
+
+Radio Board:
+
+- EFR32FG1P133F256GM48 Flex Gecko SoC
+- 8Mbit SPI NOR Flash
+
+For more information about the EFR32FG1 SoC and EFR32-SLWSTK6061A board:
```
Comment (dbkinder) [inline boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: ``board, refer to these documents:``

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/efr32_slwstk6061a/efr32_slwstk6061a_defconfig``
+
+Other hardware features are currently not supported by the port.
+
+Connections and IOs
+===================
+
+The EFR32FG1P SoC has five gpio controllers (PORTA to PORTD and PORTF). All of
```
Comment (dbkinder) [inline boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: ``GPIO``

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+them are enabled for the EFR32-SLWSTK6061A board.
+
+In the following table, the column Name contains Pin names. For example, PA2
```
Comment (dbkinder) [inline boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: ``**Name**``  (to make it bold)

**Review by dbkinder — CHANGES_REQUESTED** [review]
a few doc changes recommended

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst (hunk 1 above)
Comment (chrta) [reply boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: done

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst (hunk 2 above)
Comment (chrta) [reply boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: done

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst (hunk 3 above)
Comment (chrta) [reply boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: done

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst (hunk 4 above)
Comment (chrta) [reply boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: done

**Review by galak — COMMENTED** [review]
Any plans for a board code that uses this soc?

**Comment by chrta** [comment]
> Any plans for a board code that uses this soc?

Basic support for one board is already included (without radio functionality, only gpio and uart), see https://github.com/zephyrproject-rtos/zephyr/pull/6841/commits/cf776eb21777879b53994d23b2baa98fb8410d39

The radio support will have to wait a few month until Silabs releases their radio API under an open license.

File: boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+The EFR32 Flex Gecko Wireless Starter Kit SLWSTK6061A contains a Wireless
+System-On-Chip from the EFR32FG family built on an ARM® Cortex®-M4F processor
+with excellent low power capabilities.
+
+.. figure:: efr32-flex-gecko-starter-kit.png
+   :width: 375px
+   :align: center
+   :alt: EFR32-SLWSTK6061A
+
+   EFR32-SLWSTK6061A (image courtesy of Silicon Labs)
```
Comment (chrta) [inline boards/arm/efr32_slwstk6061a/doc/efr32_slwstk6061a.rst]: I added this note below the board image to credit Silicon Labs, see https://www.silabs.com/community/general-interest/forum.topic.html/usage_of_board_pictu-ydFZ

**Review by dbkinder — APPROVED** [review]
Thanks!

**Comment by chrta** [comment]
@galak @MaureenHelm Is there anything blocking this or is there anything i can/must do to get this merged?

File: drivers/gpio/gpio_gecko.c [hunk 6]
```diff
@@ -154,13 +154,21 @@ static int gpio_gecko_write(struct device *dev,
 			 * Writing zeros to the other bits leaves the data
 			 * output unchanged for the other pins.
 			 */
+#if defined(_GPIO_P_DOUTSET_MASK)
 			gpio_base->DOUTSET = BIT(pin);
+#else
+			BUS_RegBitWrite(&gpio_base->DOUT, pin, 1);
```
Comment (diegosueiro) [inline drivers/gpio/gpio_gecko.c]: I think you can use a more clear assignment like:
gpio_base->DOUT |= BIT(pin);

File: drivers/gpio/gpio_gecko.c [hunk 7]
```diff
@@ -154,13 +154,21 @@ static int gpio_gecko_write(struct device *dev,
...
+#endif
 		} else {
 			/* Clear the data output for the corresponding pin.
 			 * Writing zeros to the other bits leaves the data
 			 * output unchanged for the other pins.
 			 */
+#if defined(_GPIO_P_DOUTCLR_MASK)
 			gpio_base->DOUTCLR = BIT(pin);
+#else
+			BUS_RegBitWrite(&gpio_base->DOUT, pin, 0);
```
Comment (diegosueiro) [inline drivers/gpio/gpio_gecko.c]: I think you can use a more clear assignment like:
gpio_base->DOUT &= ~BIT(pin);

**Comment by galak** [comment]
I think it would be good to rebase, as I think there are a few changes from upstream that this needs updating to.

**Comment by galak** [comment]
> I think it would be good to rebase, as I think there are a few changes from upstream that this needs updating to.

I went ahead and rebased this and fixed up the few Kconfig related issues.

**Comment by chrta** [comment]
Great, thank you

**Pull request merged**